Makefile

```diff
@@ -115,31 +118,38 @@ lang:
 	@echo "Generating: lang..."
 	$(MAKE) --quiet -C lang
 
-$(PROGRAM): $(GO_FILES)
-	@echo "Building: $(PROGRAM), version: $(SVERSION)..."
-	time go build -i -ldflags "-X main.program=$(PROGRAM) -X main.version=$(SVERSION)" -o $(PROGRAM) $(BUILD_FLAGS);
-
-define buildrelease
-GOOS=$(1) GOARCH=$(2) go build -ldflags "-X main.program=$(PROGRAM) -X main.version=$(SVERSION) -s -w" -o $(PROGRAM)-$(1)-$(2) $(BUILD_FLAGS);
-endef
-
-crossbuild:
-	@echo "Building: $(PROGRAM), version: $(SVERSION), arches: $(GOOSARCHES) ..."
-	$(foreach GOOSARCH,$(GOOSARCHES), $(call buildrelease,$(subst /,,$(dir $(GOOSARCH))),$(notdir $(GOOSARCH))))
+# build a `mgmt` binary for current host os/arch
+$(PROGRAM): build/mgmt-${GOHOSTOS}-${GOHOSTARCH}
+	cp $< $@
 
 $(PROGRAM).static: $(GO_FILES)
 	@echo "Building: $(PROGRAM).static, version: $(SVERSION)..."
 	go generate
 	go build -a -installsuffix cgo -tags netgo -ldflags '-extldflags "-static" -X main.program=$(PROGRAM) -X main.version=$(SVERSION) -s -w' -o $(PROGRAM).static $(BUILD_FLAGS);
 
-build: bindata lang $(PROGRAM)
+build: $(PROGRAM)
+
+# pattern rule target for (cross)building, mgmt-OS-ARCH will be expanded to the correct build
+# extract os and arch from target pattern
+GOOS=$(firstword $(subst -, ,$*))
+GOARCH=$(lastword $(subst -, ,$*))
+build/mgmt-%: $(GO_FILES) | bindata lang
```

Comment:
Is there a `mkdir build/` somewhere that is needed? (Test if it works in a clean repo.)

Reply:
go build seems to take care of it, I had a (broken) target for it at first but dropped it